formats/cbor/commonMain/src/kotlinx/serialization/cbor/Tags.kt

```diff
@@ -0,0 +1,68 @@
+package kotlinx.serialization.cbor
+
+import kotlinx.serialization.*
+
+/**
+ * Specifies that a property shall be tagged and serialized as CBOR major type 6: optional semantic tagging
+ * of other major types.
+ *
+ * Example usage:
+ *
+ * ```
+ * @Serializable
+ * data class Data(
+ *     @ValueTags(1337uL)
+ *     @ByteString
+ *     val a: ByteArray, // CBOR major type 6 1337(major type 2: a byte string).
+ *
+ *     @ValueTags(1234567uL)
+ *     val b: ByteArray  // CBOR major type 6 1234567(major type 4: an array of data items).
+ * )
+ * ```
+ *
+ * See [RFC 8949 3.4. Tagging of Items](https://datatracker.ietf.org/doc/html/rfc8949#name-tagging-of-items).
+ */
+@SerialInfo
+@Target(AnnotationTarget.PROPERTY)
+@ExperimentalSerializationApi
+public annotation class ValueTags(@OptIn(ExperimentalUnsignedTypes::class) vararg val tags: ULong) {
```

Comment:
Generally, it is recommended to propagate opt-in in the libraries (see here: https://kotlinlang.org/docs/opt-in-requirements.html#propagating-opt-in and here: https://kotlinlang.org/docs/jvm-api-guidelines-backward-compatibility.html#the-requiresoptin-annotation). The problem here is, while `ULong` is already stable, `ULongArray` which is automatically inserted by `vararg` is not. 

I'm also not sure if it is very common for a property to have multiple tags. Have you considered using `@Repeatable` annotation instead of vararg (it works only with Kotlin 1.9.0 though, because of the #2099) or not providing such functionality at all?
I don't mind propagating ExperimentalUnsignedTypes, however, if this is necessary.

Reply:
Interesting! To be frank, I did not even think about`@Repeatable`, because `vararg` seemed intuitive to use with minimum boilerplate when defining serializable data types using multiple tags, as the declaration directly reflects what's happening in the resulting CBOR byte string.

We discussed this internally and neither of us has a string opinion on either way of providing this feature. So if `@Repeatable` is preferred, we'll, of course, go with that!

Reply:
After starting to refactor `KeyTags` and `ValueTags`  to `@Repeatable`, I notices that this would make `KeyTags`  and `ValueTags` behave differently from the `@CborArray` annotation, which requires the vararg syntax to make sense. IMO this change would therefore only make tagging inconsistent, resulting in API inconsistencies and more diverse (and hence a little more complex) code internally.

Therefore, I am now against this change. Of course, if you still prefer it we'll comply.